src/Stratis.StratisDnsD/Program.cs

```diff
@@ -41,8 +41,7 @@ public static void Main(string[] args)
         {
             try
             {
-                Network network = args.Contains("-testnet") ? Network.StratisTest : Network.StratisMain;
-                NodeSettings nodeSettings = new NodeSettings(network, ProtocolVersion.ALT_PROTOCOL_VERSION, args:args, loadConfiguration:false);
+                NodeSettings nodeSettings = new NodeSettings(protocolVersion:ProtocolVersion.ALT_PROTOCOL_VERSION, args:args, loadConfiguration:true);
```

Comment:
I still don't understand the point of having the `loadConfiguration` here, since we're calling it in FullnodeBuilder.Build anyway.
```
// Load configuration file
this.NodeSettings?.LoadConfiguration(this.Features.FeatureRegistrations);
```

Reply:
It allows us to do:

`if (nodeSettings.ConfigReader.GetOrDefault<bool>("dnsfullnode", false))`

before we get to that line of code.

Reply:
We could do away with the loadConfiguration flag entirely and proceed as follows:
- always try (it may not exist) to load the configuration from inside the constructor
- in FullnodeBuilder.Build, if we detect the file does not exist then create it only (i.e. don't load it until relaunch).

The simplification forces us to assume that the default configuration file will never contain non-commented settings so that it is always equivalent to an empty or non-existent file.

The current approach allows the default configuration to actually contain actionable settings although that is not how we currently use it.

Reply:
yep, I think we should try to load the base configuration by default.

Reply:
I'll accept this PR, please investigate what can be done to make loading the config more streamlined.
Thanks!

Reply:
Thank you. See https://github.com/stratisproject/StratisBitcoinFullNode/pull/1431.